### ISSUE TITLE
Overcoming Catastrophic Forgetting by Neuron-level Plasticity Control

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/phd-bib#31]
## [Overcoming Catastrophic Forgetting by Neuron-level Plasticity Control](https://arxiv.org/abs/1907.13322)
*To address the issue of catastrophic forgetting in neural networks, we propose a novel, simple, and effective solution called neuron-level plasticity control (NPC). While learning a new task, the proposed method preserves the knowledge for the previous tasks by controlling the plasticity of the network at the neuron level. NPC estimates the importance value of each neuron and consolidates important \textit{neurons} by applying lower learning rates, rather than restricting individual connection weights to stay close to certain values. The experimental results on the incremental MNIST (iMNIST) and incremental CIFAR100 (iCIFAR100) datasets show that neuron-level consolidation is substantially more effective compared to the connection-level consolidation approaches.*

### Cite this paper
```bibtex
@article{paik2019overcoming,
  title={Overcoming Catastrophic Forgetting by Neuron-level Plasticity Control},
  author={Paik, Inyoung and Oh, Sangjun and Kwak, Tae-Yeong and Kim, Injung},
  journal={arXiv preprint arXiv:1907.13322},
  year={2019}
}
```

**Comment by juanmc2005** [comment]
### Description

This paper presents a method to address catastrophic forgetting by determining importance at the neuron level, in contrast to most approaches that concentrate on weight importance.

They explain the problem of regularizing by weight importance by showing how it is possible that unimportant weights reach important neurons, hence modifying important weights down the line.

![image](https://user-images.githubusercontent.com/5760780/79209904-50ff3580-7e44-11ea-856a-4839d6a12fa3.png)

### Method

Their method is based on calculating the importance of a neuron according to its gradient, and then using that to introduce a learning rate loss. There is one separate learning rate per neuron that is used to update all weights in the previous layer leading to it.

They argue this is a simpler way of controlling plasticity, as it does not require to keep a copy of the consolidated weights in memory, it only relies on neuron activations and their gradients, which are already calculated during backpropagation.

### Experiments

They perform experiments in an incremental class setting on MNIST and CIFAR-100, using a simple model of 3 convolutional layers and 2 fully connected layers.
They split MNIST into 5 subsets of 2 classes each, and follow a similar splitting of CIFAR-100 with 10 subsets.

Something that is not quite "continual" in this setting is that they train each task for multiple epochs, probably to compensate for the fact that they don't use any replay mechanism.
For example, on MNIST, each task has 12000 examples, and they redefine an epoch to be 5 times each task (so the number of training data equals the total size of MNIST), and finally they train for 30 epochs on each subset. That means that a single MNIST example is seen 150 times! This is huge compared to other continual learning experiments that only see examples once.
Moreover, it is not clear whether each task is trained in an i.i.d. fashion.
In my view, this makes the experiments too simple w.r.t. to other continual learning studies.

### Results

In their results, they find that their approach outperforms a weight-focused version of their algorithm (that is not described), as well as EWC, MAS, Synaptic Intelligence, and others.
They also compare against a "Baseline (fine-tune)" that is not described in the rest of the paper, so it is not clear what is the approach taken as baseline.

### Takeaway

No immediate takeaways for our work, although the idea of concentrating regularization at the neuron level for plasticity is very interesting, Although it appears that their method is superior in this experimental setting, I don't think the experiments are convincing enough for more challenging continual learning settings and datasets.
